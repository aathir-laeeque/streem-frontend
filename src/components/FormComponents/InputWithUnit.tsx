import React, { FC } from 'react';

const InputWithUnit: FC = () => <div>Input with unit</div>;

export default InputWithUnit;
