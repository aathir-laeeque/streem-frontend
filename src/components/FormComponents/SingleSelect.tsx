import React, { FC } from 'react';
import { Field } from 'formik';

interface SingleSelectProps {
  label: string;
  name: string;
  options: { label: string; value: string | number }[];
}

const SingleSelect: FC<SingleSelectProps> = ({ label, name, options }) => (
  <div className="form-field">
    <label className="field-label">{label}</label>
    <Field as="select" name={name} className="field-value">
      {options.map((option, index) => (
        <option value={option.value} key={index}>
          {option.label}
        </option>
      ))}
    </Field>
  </div>
);

export default SingleSelect;
