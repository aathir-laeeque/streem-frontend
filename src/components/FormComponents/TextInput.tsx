import React, { FC } from 'react';
import { useField } from 'formik';

interface TextInputProps {
  label: string;
  name: string;
  value: string;
}

const TextInput: FC<TextInputProps> = ({ label, ...props }) => {
  const [field, meta] = useField(props);

  return (
    <div>
      <label htmlFor={name}>{label}</label>
      <input {...props} {...field} />
      {meta.touched && meta.error ? (
        <div className="error">{meta.error}</div>
      ) : null}
    </div>
  );
};

export default TextInput;
