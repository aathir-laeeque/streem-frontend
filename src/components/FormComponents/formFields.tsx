import React from 'react';

export const renderInputField = ({
  input,
  type,
  label,
  placeholder,
  meta: { touched, error },
}: Record<string, any>) => (
  <div>
    {label ? <label>{label}</label> : null}
    <div>
      <input {...input} type={type} placeholder={placeholder} />
      {touched && error && <span>{error}</span>}
    </div>
  </div>
);
