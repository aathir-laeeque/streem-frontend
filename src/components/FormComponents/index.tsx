export { renderInputField } from './formFields';
// export { default as SingleSelect } from './SingleSelect';
export { Form } from './styledComponents';
