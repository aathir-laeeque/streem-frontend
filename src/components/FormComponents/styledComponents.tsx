import styled from 'styled-components';

export const Form = styled.form.attrs({})`
  height: 100%;
`;
