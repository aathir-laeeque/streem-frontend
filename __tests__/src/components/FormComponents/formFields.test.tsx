import { renderInputField } from '#components/FormComponents';
import { create, ReactTestRenderer } from 'react-test-renderer';

describe('<FormField />', () => {
  let snapshot: ReactTestRenderer;

  beforeAll(() => {
    const component = renderInputField({
      input: {},
      type: 'text',
      label: 'Form Field',
      placeholder: 'Form Field',
      meta: { touched: null, error: null },
    });
    snapshot = create(component);
  });

  it('should render and match the snapshot', () => {
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});
