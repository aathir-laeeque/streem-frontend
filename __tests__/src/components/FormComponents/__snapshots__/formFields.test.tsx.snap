// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormField /> should render and match the snapshot 1`] = `
<div>
  <label>
    Form Field
  </label>
  <div>
    <input
      placeholder="Form Field"
      type="text"
    />
  </div>
</div>
`;
